frontend/packages/data-portal/app/components/BrowseData/BrowseDataSearch.tsx

```diff
@@ -0,0 +1,39 @@
+import { InputSearch } from '@czi-sds/components'
+import { useDebouncedEffect } from '@react-hookz/web'
+import { useSearchParams } from '@remix-run/react'
+import { useState } from 'react'
+
+import { i18n } from 'app/i18n'
+
+const SEARCH_QUERY_DEBOUNCE_TIME_MS = 500
+
+export function BrowseDataSearch() {
+  const [searchParams, setSearchParams] = useSearchParams()
+  const [query, setQuery] = useState(searchParams.get('search') ?? '')
+
+  useDebouncedEffect(
+    () =>
+      setSearchParams((prev) => {
+        if (query) {
+          prev.set('search', query)
+        } else if (prev.has('search')) {
+          prev.delete('search')
+        }
+
+        return prev
+      }),
+    [query],
+    SEARCH_QUERY_DEBOUNCE_TIME_MS,
+  )
```

Comment:
you went over why we need a debounce in the live code review, could you maybe add a comment somewhere here in case I forget later down the line?